### PR TITLE
[profiler] Remove usage of onEachDevice from legacy profiler

### PR DESCRIPTION
**Pull request by ilia-cher**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#54125 Remove usage of onEachDevice from legacy profiler**

Summary:
Fixes https://github.com/pytorch/pytorch/issues/48987

Test Plan:
python setup.py clean
TORCH_CUDA_ARCH_LIST="6.0" USE_CUDA=1 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake 2>&1 | tee ~/output.txt
python test/test_profiler.py -v

python setup.py clean
USE_CUDA=0 USE_MKLDNN=1 BLAS=MKL BUILD_BINARY=1 python setup.py develop install --cmake 2>&1 | tee ~/output.txt
python test/test_profiler.py -v

+ CI

Differential Revision: [D27109481](https://our.internmc.facebook.com/intern/diff/D27109481)

### CODE REVIEW

File: torch/csrc/distributed/rpc/utils.cpp [hunk 1]
```diff
@@ -520,79 +522,50 @@ std::vector<at::IValue> readWrappedPayload(
...
-        e.setCudaUs(cudaUs);
+        if (e.kind() == EventKind::PushRange) {
+          startEvents[e.handle()] = &e;
+          e.setCudaUs(0);
+        } else if (e.kind() == EventKind::PopRange) {
+          auto it = startEvents.find(e.handle());
+          if (it != startEvents.end()) {
+            e.setCudaUs(it->second->cudaElapsedUs(e));
+          } else {
+            TORCH_WARN("Found a pop event without a corresponding push event");
```
Comment (rohan-varma) [inline torch/csrc/distributed/rpc/utils.cpp:566]: nit: might be good to log the event to help debugging. Also, should we `setCudaUs` to some reasonable value here?

**Review by rohan-varma — APPROVED** [review]
LGTM, but RPC test is failing, can help take a look at that

File: torch/csrc/distributed/rpc/utils.cpp (hunk 1 above)
Comment (ilia-cher) [reply torch/csrc/distributed/rpc/utils.cpp:566]: sg

Comment (ilia-cher) [reply torch/csrc/distributed/rpc/utils.cpp:566]: will set `setCudaUs`, also we don't have names in pop events

**Comment by ilia-cher** [comment]
fixed RPC test issue

**Pull request closed without merging**